cmd/dependencyverifier/dependencyverifier.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// runCommand runs the cmd and returns the combined stdout and stderr, or an
+// error if the command failed.
+func runCommand(cmd ...string) (string, error) {
+	output, err := exec.Command(cmd[0], cmd[1:]...).CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to run %q: %s (%s)", strings.Join(cmd, " "), err, output)
+	}
+	return string(output), nil
+}
+
+func readFile(path string) (string, error) {
+	content, err := ioutil.ReadFile(path)
+	// Convert []byte to string and print to screen
+	return string(content), err
+}
+
+func getDependencies(dependenciesStr, key string) map[string]string {
+	var dependencies map[string]map[string]string
+	json.Unmarshal([]byte(dependenciesStr), &dependencies)
+	return dependencies[key]
+}
+
+func convertToMap(modStr string) map[string]string {
```

Comment:
this doesn't look quite right... the left side is the parent, the right side is the dependent... returning map[string]string means that we can only express a single parent or a single child for a given module

I think we want to return `map[string][]string`, where the key is the child module (the right side of `go mod graph` output) and the value is the list of module names that refer to it

Reply:
Updated.